gmm_client/src/client_utils.erl

```diff
@@ -0,0 +1,135 @@
+%%%-------------------------------------------------------------------
+%% @doc
+%%  Basic functions to switch between JSON and erlang types
+%% @end
+%%%-------------------------------------------------------------------
+
+-module(client_utils).
+
+%% API
+-export([
+    empty_json/0,
+    encode/1,
+    decode/1,
+
+    parse_boolean/1,
+    validate_vertex_id/1,
+    validate_edge_id/1,
+
+    split_bin/2,
+    split_bin/1,
+
+    create_vertex_id/2,
+    split_vertex_id/1,
+    owner_of/1,
+
+    create_edge_id/2,
+    split_edge_id/1,
+
+    lists_equal/2
+]).
+
+
+%%%---------------------------
+%% JSON manipulation
+%%%---------------------------
+
+-spec empty_json() -> binary().
+empty_json() -> jiffy:encode({[]}).
+
+-spec encode(any()) -> binary().
+encode(Val) -> jiffy:encode(Val).
+
+-spec decode(binary()) -> any().
+decode(Val) -> jiffy:decode(Val, [return_maps]).
+
+
+%%%---------------------------
+%% Argument's validation
+%%%---------------------------
+
+-spec parse_boolean(binary()) -> {ok, boolean()} | {error, any()}.
```

Comment:
To mi nie wygląda na funkcję parsującą coś co przychodzi od użytkownika, więc ja bym bardziej sugerował zrobić coś w tym stylu i niech to rzuca badarg jak dostanie coś czego się nie da sparsować na boolean
```suggestion
-spec parse_boolean(binary()) -> boolean()
```

Reply:
Hmm.. w zasadzie to JEST funkcja parsująca coś przychodzące od użytkownika. Dokładniej rzecz ujmując, to przekopiowałem tę funkcję z gmm_utils. Kwestia jest taka, że serwer powinien spodziewać się śmieci od klienta, ale klient nie musi się spodziewać śmieci od serwera